### PR TITLE
Plugin should throw same arithmetic exceptions as Spark part1

### PR DESCRIPTION
**Pull request by HaoYang670**
This is the first PR for #5196.
In this PR we:
   1. move `RapidsErrorUtils` from `311until330-nondb` to `311until320-nondb` and `320until330-nondb`.
   2. Add `divByZeroError` and `divOverflowError` into `RapidsErrorUtils`
   3. update some tests
  
There are other arithmetic exceptions that we need to update. I will file more follow-up PRs to fix them.

### CODE REVIEW

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
> If there are a number of followup items, it may be useful to enumerate what needs to be done in #5196 as a checklist so it's clear what has been done and what is yet to be done.

Done! https://github.com/NVIDIA/spark-rapids/issues/5196#issue-1200009075

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala [hunk 1]
```diff
@@ -724,16 +716,16 @@ trait GpuDivModLike extends CudfBinaryArithmetic {
...
         if (checkDivideOverflow && isDivOverflow(lhs, rhs)) {
-          divOverflowError()
+          throw RapidsErrorUtils.divOverflowError(origin)
         }
         super.doColumnar(lhs, rhs)
       }
     } else {
       if (checkDivideOverflow && isDivOverflow(lhs, rhs)) {
-        divOverflowError()
+          throw RapidsErrorUtils.divOverflowError(origin)
```
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala:728]: Indentation seems off here
```suggestion
        throw RapidsErrorUtils.divOverflowError(origin)
```

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala [hunk 2]
```diff
@@ -743,7 +735,7 @@ trait GpuDivModLike extends CudfBinaryArithmetic {
 
   override def doColumnar(lhs: GpuScalar, rhs: GpuColumnVector): ColumnVector = {
     if (checkDivideOverflow && isDivOverflow(lhs, rhs)) {
-      divOverflowError()
+          throw RapidsErrorUtils.divOverflowError(origin)
```
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala:738]:
```suggestion
      throw RapidsErrorUtils.divOverflowError(origin)
```

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala [hunk 3]
```diff
@@ -753,15 +745,15 @@ trait GpuDivModLike extends CudfBinaryArithmetic {
   override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
     if (isScalarZero(rhs.getBase)) {
       if (failOnError) {
-        divByZeroError()
+          throw RapidsErrorUtils.divByZeroError(origin)
```
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala:748]:
```suggestion
        throw RapidsErrorUtils.divByZeroError(origin)
```

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala [hunk 4]
```diff
@@ -753,15 +745,15 @@ trait GpuDivModLike extends CudfBinaryArithmetic {
...
       } else {
         withResource(Scalar.fromNull(outputType(lhs.getBase, rhs.getBase))) { nullScalar =>
           ColumnVector.fromScalar(nullScalar, lhs.getRowCount.toInt)
         }
       }
     } else {
       if (checkDivideOverflow && isDivOverflow(lhs, rhs)) {
-        divOverflowError()
+          throw RapidsErrorUtils.divOverflowError(origin)
```
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala:756]:
```suggestion
        throw RapidsErrorUtils.divOverflowError(origin)
```

**Comment by HaoYang670** [comment]
build

**Review by jlowe — DISMISSED** [review]

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
build

**Comment by jlowe** [comment]
build

**Comment by HaoYang670** [comment]
Why is there such an error `/home/ubuntu/spark-rapids/sql-plugin/src/main/320until330-noncdh/scala/com/nvidia/spark/rapids/shims/OrcShims.scala:22: not found: type OrcShims320untilAllBase`? 
Do I modify the `pom.xml` in a wrong way?
https://github.com/NVIDIA/spark-rapids/runs/6269757840?check_suite_focus=true#step:2:3836

**Comment by jlowe** [comment]
> Why is there such an error

It's unrelated to this PR.  The problem was introduced by #5346 and fixed by #5414.

**Comment by jlowe** [comment]
build

**Comment by HaoYang670** [comment]
Could we merge this PR, as the potential merge conflicts are high?

**Review by jlowe — APPROVED** [review]

**Pull request merged**